ceph/rados/core_workflows.py

```diff
@@ -1034,18 +1035,206 @@ def get_journalctl_log(
+        Args:
+            flag: Rados flag to set
+            Returns : True -> pass, False -> fail
+        """
+        pgid_cmd = "ceph pg dump"
+        pgid_list = []
+        try:
+            pg_dump = self.run_ceph_command(cmd=pgid_cmd)
+            pg_dump_json = json.dumps(pg_dump)
+            pg_load_json = json.loads(pg_dump_json)
+            for pg in pg_load_json["pg_map"]["pg_stats"]:
+                pgid_list.append(pg["pgid"])
+            print("The pg list is :::::", pgid_list)
+            return pgid_list
+        except Exception:
+            log.error("Error while un-setting the flag")
+            return False
+
+    def start_all_pg_scrub(self):
+
+        base_scrub_pg_cmd = "ceph pg scrub  "
+        try:
+            pg_list = self.get_pgid_list()
+            for pg in pg_list:
+                scrub_pg_cmd = base_scrub_pg_cmd + pg
+                self.run_ceph_command(cmd=scrub_pg_cmd)
+            return True
+        except Exception:
+            log.error("Error while un-setting the flag")
+            return False
+
+    def start_all_pg_deepscrub(self):
 
-        ceph config set osd_mclock_profile <profile_name>
+        base_deepscrub_pg_cmd = "ceph pg deep-scrub  "
+        try:
+            pg_list = self.get_pgid_list()
+            for pg in pg_list:
+                scrub_pg_cmd = base_deepscrub_pg_cmd + pg
+                self.run_ceph_command(cmd=scrub_pg_cmd)
+            return True
+        except Exception:
+            log.error("Error while un-setting the flag")
+            return False
 
-        profile names:
-        - balanced
-        - high_recovery_ops
-        - high_client_ops
+    def get_pg_states(self):
+        """
+        Used to get the pg states
+        """
+        cmd = "ceph -s -f json"
+        pgstate_list = []
+        try:
+            output = self.run_ceph_command(cmd=cmd)
+            output_dump = json.dumps(output)
+            output_load_json = json.loads(output_dump)
+            for pg in output_load_json["pgmap"]["pgs_by_state"]:
+                pgstate_list.append(pg["state_name"])
+            return pgstate_list
+        except Exception:
+            log.error("Error while un-setting the flag")
+            return False
 
+    def get_osd_list(self):
+        """
+        Used to get the list of OSDs configured in the cluster nodes
+        Returns:
+            Dictionary with the keys as cluster osd names and values are the list of osd names configured to the node.
+        """
+        try:
+            osd_list = dict()
+            cmd = "ceph osd  tree"
+            osds = self.run_ceph_command(cmd)
+            for entry in osds["nodes"]:
+                if entry["type"] == "host":
+                    osd_list[entry["name"]] = entry["children"]
+            return osd_list
+        except Exception as e:
+            log.error("Failed to  get the osd list : %s", str(e))
+
+    def get_osd_pid(self, host_object, osd):
```

Comment:
can we have generic pid getter method which could be used for any ceph process. using its binary path from process list.